invenio_files_rest/serializer.py

```diff
@@ -18,6 +19,44 @@
 from .models import Bucket, MultipartObject, ObjectVersion, Part
 
 
+class MarshalResultBase(object):
+    """Wrapper class to provide backward compatibility with marshmallow 2."""
+
+    def __new__(cls, result, errors=None, *args, **kwargs):
+        """Instanciate Marshal Result class."""
+        def data(self):
+            warnings.warn(
+                "Schema().dump().data attribute is deprecated "
```

Comment:
Will this issue the warning on Marshmallow 2? 

I think the warning should only be issued when v3 is installed to warn users that their schema will break at some point.

I guess this class is also useful for other modules. Perhaps we need to find another home for it?